pkg/sbproxy/reconcile_brokers.go

```diff
@@ -181,6 +185,28 @@ func (r ReconcileBrokersTask) deleteBrokerRegistration(broker *platform.ServiceB
 	}
 }
 
+func (r ReconcileBrokersTask) enableServiceAccessVisibilities(broker *platform.ServiceBroker) {
+	if f, isEnabler := r.platformClient.(platform.ServiceAccess); isEnabler {
+		emptyContext := emptyContext()
+		logrus.WithFields(logBroker(broker)).Info("ReconcileBrokersTask task attempting to enable service access for broker...")
+
+		catalog := broker.Catalog
+		if catalog == nil {
+			logrus.WithFields(logBroker(broker)).Error("Error enabling service access due to missing catalog details")
+			return
+		}
+
+		for _, service := range catalog.Services {
```

Comment:
Just a thought - should any of these "enable-service-access" and/or "de/register-service-broker" be in go-routines?

Reply:
it would make sense to some extend - but all register/deregister broker go routines would need to have finished before we start any enable (and in the future disable) service access routines. Also, this reconcilation right now happens in a go routine (not that its a problem, but it may lead to some complications)